### PR TITLE
Fix retry-queue's enqueue() blocked by semaphore

### PR DESCRIPTION
**Pull request by ezdac**
Before, the retry-queues semaphore had to be acquired in order to
enqueue messages.
This caused the thought to be asynchronous enqueue() method and
all methods calling it to be dependent on successfully processing
all messages in the queues greenlet that processes the queue.

Effectively this resulted in a synchronous enqueue() call and could
cause the state-machine to block during long-running requests or retries
to the transport layer.

Fixes: #7120

### CODE REVIEW

File: raiden/network/transport/matrix/transport.py [hunk 1]
```diff
@@ -361,13 +350,19 @@ def _run(self) -> None:  # type: ignore
...
+
+            self._notify_event.clear()
+            if self._message_queue:
+                self._idle_since = 0
+                # Shallow copy the current messages in the queue
+                # to be processed for this _check_and_send iteration
+                messages_current_workload = self._message_queue[:]
+                # then empty the message_queue
+                self._message_queue.clear()
+                with self._lock:
```
Comment (istankovic) [inline raiden/network/transport/matrix/transport.py:362]: Is this lock really necessary? `_check_and_send` is only called from `_run` and I can't see how we could end up calling `_check_and_send` on the same `RetryQueue` twice. Note that the lock is _per-RetryQueue-instance_.

Somewhat unfortunate, though, is that we manually call `_check_and_send` in `raiden/tests/mocked/test_matrix_transport.py`, but I'm hoping we could get around that.

Comment (ezdac) [reply raiden/network/transport/matrix/transport.py:362]: Yes, it is very likely unnecessary, but I wanted to be careful for now.

Comment (ezdac) [reply raiden/network/transport/matrix/transport.py:362]: And it could also be helpful to avoid later programming errors.

Comment (ezdac) [reply raiden/network/transport/matrix/transport.py]: The lock now was entirely removed

**Review by ulope — APPROVED** [review]
LGTM

**Pull request merged**